lib/cloudregister/registerutils.py

```diff
@@ -516,6 +517,90 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def setup_registry(registry_fqdn, username, password):
+    """Set all the necessary parts for the registry,
+       returns True if the setup completed, False otherwise."""
+    if not os.path.exists(os.path.dirname(REGISTRY_CREDENTIALS_PATH)):
+        os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH))
+
+    setup_registry_succeed = set_registry_auth_token(
+        registry_fqdn, username, password
+    )
+    return setup_registry_succeed
+
+
+# ----------------------------------------------------------------------------
+def get_registry_credentials():
+    """Read the registry credentials file
+       and return its content or an empty dict."""
+    config_json = {}
+    failed = False
+    if os.path.exists(REGISTRY_CREDENTIALS_PATH):
+        try:
+            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+                config_json = json.load(cred_json)
+        except json.decoder.JSONDecodeError:
```

Comment:
We do not catch open errors this way, intentionally ?

Reply:
The path exists, we are root. 2 issues remain, the system is out of file handles or the disk is bad. For neither there is anything we can do about it, It's OK to blow up.

Reply:
Done in [93bf1aa](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/159/commits/93bf1aa8e92b4b7acee4aab51034087fbeb84149)